disnake/state.py

```diff
@@ -952,7 +963,7 @@ def parse_interaction_create(self, data: gateway.InteractionCreateEvent) -> None
             self.dispatch("message_interaction", interaction)
             if interaction.data.component_type is ComponentType.button:
                 self.dispatch("button_click", interaction)
-            elif interaction.data.component_type is ComponentType.select:
+            elif interaction.data.component_type in _SELECT_COMPONENT_TYPES:
```

Comment:
this makes this change breaking, because on_dropdown now takes a BaseSelectMenu instead of SelectMenu (and that documentation change should be made)

Reply:
`on_dropdown` receives the `interaction` object, so this isn't a breaking change.
If the component was passed directly, I'd still argue it wouldn't be breaking, since getting an unexpected component type would require the bot to send such a component in the first place.

Reply:
right